kernel/nsh/nsh.c

```diff
@@ -0,0 +1,150 @@
+// SPDX-FileCopyrightText: CGL-KFS
+// SPDX-License-Identifier: BSD-3-Clause
+
+/* kernel/nsh/nsh.c
+ *
+ * Nulix shell
+ *
+ * created: 2022/12/07 - xlmod <glafond-@student.42.fr>
+ * updated: 2022/12/09 - lfalkau <lfalkau@student.42.fr>
+ */
+
+#include <kernel/string.h>
+#include <kernel/print.h>
+#include <kernel/screenbuf.h>
+#include <kernel/nsh.h>
+#include <kernel/keyboard.h>
+#include <kernel/builtins.h>
+
+extern struct screenbuf sb[];
+extern int sb_index;
+
+char nsh_buf[NSH_BUFSIZE];
+char *nsh_cmd[NSH_BUFSIZE / 2];
+int nsh_cmdnarg;
+int nsh_bufindex;
+
+struct builtin builtin[] = {
+	{"reboot", reboot},
+	{"poweroff", poweroff},
+	{"exit", poweroff},
+	{"keymap", keymap},
+	{"color", color},
+	{"alloc", alloc},
+	{"free", free},
+	{"info", info},
+	{"clear", clear},
+	{"hexdump", hexdump},
+	{"next", next},
+	{"prev", prev},
+	{"help", help},
+	{"interrupt", interrupt},
+	{NULL, NULL},
+};
+
+
+static void nsh_prompt() {
+	uint8_t oldcolor;
+
+	oldcolor = sb_get_color(sb + sb_index);
+		sb_set_fg(sb + sb_index, SB_COLOR_LIGHT_BLUE);
+	kprintf("nsh$ ");
+	sb_set_color(sb + sb_index, oldcolor);
+}
```

Comment:
Wow, this looks very nice! This guy codes very well...

Reply:
mdr